java/src/lib.rs

```diff
@@ -19,18 +20,50 @@ pub use ffi_test::*;
 fn redis_value_to_java<'local>(env: &mut JNIEnv<'local>, val: Value) -> JObject<'local> {
     match val {
         Value::Nil => JObject::null(),
-        Value::SimpleString(str) => JObject::from(env.new_string(str).unwrap()),
+        Value::SimpleString(str) => {
+            let jstr = JObject::from(env.new_string(str).unwrap());
+
+            // TODO cache class
+            let class = Desc::<JClass>::lookup("glide/api/models/GlideString", env).unwrap();
+
+            JObject::try_from(
+                env.call_static_method(
+                    class,
+                    "of",
+                    "(Ljava/lang/String;)Lglide/api/models/GlideString;",
+                    &[(&jstr).into()],
+                )
+                .unwrap(),
+            )
+            .unwrap()
+
+            // TODO drop(class)
+        }
         Value::Okay => JObject::from(env.new_string("OK").unwrap()),
         Value::Int(num) => env
             .new_object("java/lang/Long", "(J)V", &[num.into()])
             .unwrap(),
-        Value::BulkString(data) => match std::str::from_utf8(data.as_ref()) {
-            Ok(val) => JObject::from(env.new_string(val).unwrap()),
-            Err(_err) => {
-                let _ = env.throw("Error decoding Unicode data");
-                JObject::null()
-            }
-        },
+        Value::BulkString(data) => {
+            let array = env.new_byte_array(data.len() as i32).unwrap();
+            let i8vec: Vec<i8> = data.into_iter().map(|b| b as i8).collect();
+            env.set_byte_array_region(&array, 0, &i8vec[..]).unwrap();
+
+            // TODO cache class
+            let class = Desc::<JClass>::lookup("glide/api/models/GlideString", env).unwrap();
+
+            JObject::try_from(
+                env.call_static_method(
```

Comment:
How expensive is this call? (I don't see alternative for doing this, I wonder what's the impact of these calls)